src/fromager/bootstrapper.py

```diff
@@ -88,11 +88,28 @@ def resolve_version(
             return self._resolved_requirements[req_str]
 
         pbi = self.ctx.package_build_info(req)
-        if pbi.pre_built:
-            source_url, resolved_version = self._resolve_prebuilt_with_history(
-                req=req,
-                req_type=req_type,
-            )
+
+        # Check if package has version-specific settings (any version-specific config)
+        variant_info = pbi._ps.variants.get(pbi._variant)
+        has_version_specific_prebuilt = (
```

Comment:
Oh, this is going to be messier than I thought.

The `_ps` field really should be private to the PBI. We could at least wrap this logic in a method, but it's odd to need to know about the version-specific prebuilt flags before we resolve the version we're going to bootstrapping.

I don't remember why we have different logic for resolving versions of prebuilt wheels. How different is that logic from the other flow? Does it skip some steps or are there completely different steps?